ci/scripts/python_wheel_macos_build.sh

```diff
@@ -129,9 +131,9 @@ cmake \
     -DCMAKE_INSTALL_PREFIX=${build_dir}/install \
     -DCMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES} \
     -DCMAKE_UNITY_BUILD=${CMAKE_UNITY_BUILD} \
-    -DOPENSSL_USE_STATIC_LIBS=ON \
```

Comment:
This seems unrelated to the other changes. Can you give a short reasoning for it?

Reply:
Yes. This is unrelated to `ARROW_PYTHON`. Sorry for including this to this pull request.

`OPENSSL_USE_STATIC_LIBS` is redundant here because there is `-DARROW_DEPENDENCY_USE_SHARED=OFF` in this command line. `OPENSSL_USE_STATIC_LIBS` is set automatically in https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindOpenSSLAlt.cmake#L48-L52 .

I just found this by sorting CMake options. So I mix this change into this pull request. Sorry.

Reply:
No need to be sorry. I was just wondering about the reasoning